DropboxSync.BLL/Configurations/UploadConfiguration.cs

```diff
@@ -0,0 +1,25 @@
+﻿using DropboxSync.BLL.Entities;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Metadata.Builders;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace DropboxSync.BLL.Configurations
+{
+    public class UploadConfiguration : IEntityTypeConfiguration<UploadEntity>
+    {
+        public void Configure(EntityTypeBuilder<UploadEntity> builder)
+        {
+            builder.HasKey(d => d.Id);
+
+            builder.Property(d => d.DropboxPath)
+                .IsRequired();
+
+            builder.Property(d => d.BackUpPath)
```

Comment:
what is BackupUpPath? is it when a file is logically deleted?

Reply:
BackUp path represent the local path to the file in the project in case an error occured in Dropbox, the file can still be retrieved without loss

Reply:
also using the id instead of originalFilename will allow having files with the same same in the backup folder*

Reply:
![image](https://user-images.githubusercontent.com/56565073/166901797-9d6a8c7b-24bb-4190-9ba3-3e21c0782ce1.png)

I updated the Upload table.

As you mentionned, the absoluth path is declared by the app and the database field shouldn't depend on it.

It is instead a combination of the ID as its name and the extension.

`[PATH_FROM_ENV]\[ID].[FILE_EXTENSION]`

After a close look at Dropbox's documentation on [file creation](https://www.dropbox.com/developers/documentation/http/documentation#file_requests-create) it appears that when a file is created, Dropbox returns a json containing different informations but especially a file ID and the API route to get the file.

Hosting the file ID instead of the dropbox path would be a much better solution so the table field DropboxPath changed to DropboxFileId

To retrieve the document, a API call at url `https://www.dropbox.com/request/[DROPBOX_FILE_ID]` is enough

**NOTE:** Files saved in the backup folder have no hierarchical order, they are all stored together

Reply:
Looks good!